src/app/pages/account/users/user-form/user-form.component.ts

```diff
@@ -375,51 +349,76 @@ export class UserFormComponent {
         entityForm.formGroup.controls['shell'].setValue(res[0].shell);
         entityForm.setDisabled('password', true);
         entityForm.setDisabled('password_conf', true);
-        _.find(this.fieldConfig, {name : "password"})['isHidden'] = true;
-        _.find(this.fieldConfig, {name : "password_conf"})['isHidden'] = true;
-        _.find(this.fieldConfig, {name : "password_edit"})['isHidden'] = false;
-        _.find(this.fieldConfig, {name : "password_conf_edit"})['isHidden'] = false;
+
+        this.fieldSets
+          .hideConfig("password")
+          .hideConfig("password_conf")
+          .showConfig("password_edit")
+          .showConfig("password_conf_edit");
+
         if (res[0].builtin) {
           entityForm.formGroup.controls['uid'].setValue(res[0].uid);
           entityForm.setDisabled('uid', true);
           entityForm.setValue('group',res[0].group.id);
           entityForm.setDisabled('group',true);
           entityForm.setDisabled('home',true);
           entityForm.setDisabled('home_mode',true);
-          _.find(this.fieldConfig, {name : "home_mode"})['isHidden'] = true;
+          this.fieldSets.hideConfig("home_mode");
         } else {
           entityForm.formGroup.controls['uid'].setValue(res[0].uid);
           entityForm.setDisabled('group',false);
           entityForm.setValue('group',res[0].group.id);
           entityForm.formGroup.controls['shell'].setValue(res[0].shell);
-
         }
       } else {
         this.ws.call('user.get_next_uid').subscribe((next_uid)=>{
           entityForm.formGroup.controls['uid'].setValue(next_uid);
         })
       }
-      entityForm.ws.call('notifier.choices', [ 'SHELL_CHOICES' ]).subscribe((SHELL_CHOICES) => {
-        this.shell = _.find(this.fieldConfig, {name : "shell"});
-        this.shells = SHELL_CHOICES;
-        SHELL_CHOICES.forEach((item) => {
-        if (entityForm.isNew) {
-          if(item[1] !== "netcli.sh"){
-            this.shell.options.push({label : item[1], value : item[0]});
-            entityForm.formGroup.controls['shell'].setValue(
-            this.shells[1][0]);
-          };
-        }
-        else {
-          if(entityForm.data && !entityForm.data.bsdusr_builtin) {
-            if(item[1] !== "netcli.sh"){
-              this.shell.options.push({label : item[1], value : item[0]});
-            }
-          } else {
-            this.shell.options.push({label : item[1], value : item[0]});
-          } 
-        } 
+      this.userService.shellChoices().then(choices => {
+        this.shell = this.fieldSets.config("shell");
+
+        // Leaving this here for reference. Will delete after review/before merge.
```

Comment:
Here's where things get confusing @william-gr

Reply:
Ok, I get it know.

netcli only makes sense for builtin users (in fact, only root), so presenting that option to regular users wouldnt make sense, as they dont have privileges to use it.

We could probably handle that in middleware side if we change user.shell_choices to accept an user id as parameter

Reply:
Should I remove the filtering then? Just use all choices?

Reply:
Yes, lets filter it on middleware side. Your recent change looks good